### DISCUSSION TITLE
How to have i18n.ts come along with the build?

### DISCUSSION THREAD

**Discussion by boris-arkenaar** [amannn/next-intl#480] [Q&A] [answered]
I'm trying out the translations server side. It works in development, but I'm running into an issue when built. The build runs fine, but when I run the app afterwards it exits with the message that it can't find 'i18n.ts'. What is the best approach to have this file come along in the build?

**Comment by agustif** [comment]
Make sure you add it to your tsconfig.json?

Where are you locating/importing it, what's the error message?

**Comment by boris-arkenaar** [comment]
I'm following the docs https://next-intl-docs.vercel.app/docs/getting-started/app-router-server-components#i18nts

Didn't think much of it, but when building and then launching I get the following in the console:

```
- ready started server on [::]:3000, url: http://localhost:3000
Error: Could not find i18n config at /app/i18n.ts, please provide a valid path.
```

Looking at the code in the docs again I see that `i18n.ts` is not explicitly imported but the path is just passed as a string to the `next-intl/plugin` function. So, `i18n.ts` needs to be copied to the resulting build verbatim. I explicitly copy this file now in my Dockerfile. (Running on Vercel works without any changes though.) Is that the correct approach, or am I missing something?

**Reply by agustif** [reply]
You just need to import the file from where it is and pass it doesnt really matter the path afaik


Should work both in root and /app/ I THINK i even used it in a separate package in a monorepo to have all the i18n stuff co-located there

**Reply by boris-arkenaar** [reply]
Okay, thanks. I just have to make sure to explicitly include it in the deploy.

**Reply by boris-arkenaar** [reply]
I'm not importing the file, that's the thing. It goes like this: `const withNextIntl = require('next-intl/plugin')('./i18n.ts');`

**Reply by boris-arkenaar** [reply]
No idea if I can do something like

```
import i18n from './i18n';
const withNextIntl = require('next-intl/plugin')(i18n);
```

Can't find the source for this function in the repo.

**Reply by boris-arkenaar** [reply]
No, the function requires a string as input.

**Reply by amannn** [reply, marked as answer]
> `i18n.ts` needs to be copied to the resulting build verbatim.



That's a great point, thanks for pointing this out @boris-arkenaar! We only need the `i18n.ts` file during `next dev` and `next build`, not during `next start`.

I've addressed this in https://github.com/amannn/next-intl/commit/d4eee0d3261a0354c120b063f3905fcd7dfd395f, the next RSC beta release will no longer require the file to be present after the build.

**Reply by boris-arkenaar** [reply]
That's great to hear! Makes more sense.
